### ISSUE TITLE
[Streaming] Reproducing librispeech results - RNN-T emformer 

### ISSUE THREAD

**Issue by funboarder13920** [k2-fsa/icefall#383]
Hello,

I tried to reproduce your results on librispeech streaming: https://github.com/k2-fsa/icefall/pull/278#issuecomment-1085801325
I am not done with my hyper parameter search but I was not able to get even close to the results reported.

Do you recall the configuration you used for this training ?

Best,

**Comment by csukuangfj** [comment]
The model was training using https://github.com/k2-fsa/icefall/pull/278
with the following training commands:
```
./transducer_emformer/train.py \
  --world-size 8 \
  --num-epochs 65 \
  --start-epoch 0 \
  --exp-dir transducer_emformer/exp-full \
  --full-libri 1 \
  --max-duration 200 \
  --prune-range 5 \
  --lr-factor 5 \
  --lm-scale 0.25 \
  --master-port 12358 \
  --num-encoder-layers 18 \
  --left-context-length 128 \
  --segment-length 8 \
  --right-context-length 4
```

I am using `--epoch 52 --avg 12` for decoding and testing.

**Comment by funboarder13920** [comment]
Thank you.
Did you disable time warping in specaug ?

**Comment by csukuangfj** [comment]
> I am not done with my hyper parameter search but I was not able to get even close to the results reported.

Please use the changes from that PR directly, not the latest master, and not the latest streaming branch.

I find that https://github.com/k2-fsa/icefall/pull/358 makes the WER slightly worse.

**Comment by glynpu** [comment]
>  it was necessary to append some silence to force out the final symbols 

yes, dummy extra tailing silence indeed helps, at least for that model.  https://github.com/k2-fsa/icefall/pull/242

**Comment by danpovey** [comment]
cool!
BTW, i suspect the reason why the padding frames were degrading WER, was because we apply the padding mask in test time.  I suspect that without the padding mask, it would be more robust to trailing silence.

**Comment by ngoel17** [comment]
/k2/mutual_information.py", line 157, in forward
    ans = _k2.mutual_information_forward(px, py, boundary, p)
RuntimeError: "mutual_information_cuda_stub" not implemented for 'Half'

I am getting this error while trying to run the emformer recipe. Updated k2 to latest version. Do you think I did not update k2 correctly?
